feat(clayui.com): Improve ClayNav documentation

kresimir-coko opened the pull request:
This is my proposal for improving ClayNav documentation. It started with #3205 but after investigating the code I realised that with a little bit of work we can make the Nav docs much better.

So I've done the following:

- Removed Nav Pills page and added it as a section in Markdown/CSS of Navigation page
- Renamed Nav page to Navigation, sacrificing some technicality in order to provide clarity
- Remove Vertical Navigation page and added it as a section in React Component of Navigation
- Added simple props to Nav and NavLink, such as `justified`, `disabled`, etc.
- Added a Story with `navPills` prop

I didn't add any new tests as I think it would result in many tests that provide very little value, plus similar tests didn't already exist within the component which led me to believe they aren't worthwhile.

bryceosterhaus commented on clayui.com/content/docs/components/css-nav.md:
````diff
@@ -382,6 +384,70 @@ Add `nav-unstyled` to your nav to remove spacing around `nav-link` and `nav-btn`
 </ul>
 ```
 
+### Nav Pills
````
Going off of chema's recommendation, I think we can just remove any reference to nav-pills since its really just a bootstrap thing and not necessarily a Lexicon thing. https://github.com/liferay/clay/issues/3205#issuecomment-626547433

bryceosterhaus commented on clayui.com/content/docs/components/css-nav.md:
````diff
@@ -382,6 +384,70 @@ Add `nav-unstyled` to your nav to remove spacing around `nav-link` and `nav-btn`
+</ul>
+```
+
+### Nav Justified
````
Same here, I don't think this is a lexicon design so we can just remove it.

bryceosterhaus commented on packages/clay-nav/README.mdx:
```diff
@@ -1,17 +1,53 @@
-# clay-nav
+---
+title: 'Navigation'
```
Lets leave the name as `Nav` since that is the name of our component. Navigation could be all sorts of components

kresimir-coko commented:
Hey @bryceosterhaus I've squashed the already reviewed commits and added one on top of those containing the requested changes!

bryceosterhaus commented on packages/clay-nav/src/Nav.tsx:
```diff
@@ -11,32 +11,60 @@ import {NavLink} from './NavLink';
-	 * Flag to indicate if `nav-nested` class should be applied.
+	 * Flag to indicate if `nav-pills` class should be applied. Adds background colors and border styles to navigation items.
+	 */
+	navPills?: boolean;
```
I think you might've missed these props

kresimir-coko replied:
Oh, you gave a 👍 to that [here](https://github.com/liferay/clay/pull/3207#pullrequestreview-409504327) so I thought those were okay, sure, I'll remove them.

bryceosterhaus commented on packages/clay-nav/src/Nav.tsx:
```diff
@@ -11,32 +11,60 @@ import {NavLink} from './NavLink';
+	/**
+	 * Flag to indicate if `nav-justified` class should be applied. Make navigation items equal width relative to its parent container.
+	 */
+	justified?: boolean;
```
^

bryceosterhaus commented on packages/clay-nav/src/Nav.tsx:
```diff
@@ -11,32 +11,60 @@ import {NavLink} from './NavLink';
+	/**
+	 * Flag to indicate if `nav-unstyled` class should be applied. Removes spacing around nav-link and nav-btn.
+	 */
+	unstyled?: boolean;
```
^

kresimir-coko commented:
Hey @bryceosterhaus I've removed the 3 props I added. There was a small misunderstanding which resulted in me leaving them the last time. 

One thing to note, this can be done in a future task if we wanna change it, but we already had nav-unstyled documented [here](https://clayui.com/docs/components/css-nav.html#nav-unstyled), which is what prompted me to add docs for other things too.

I didn't remove the documentation on the mentioned nav-unstyled, but we should decide on whether to have none or all of the CSS/Lexicon/Bootstrap options documented in components like this one.
